library/src/hcc_detail/hipsolver.cpp

```diff
@@ -5642,6 +5694,904 @@ catch(...)
     return exception2hip_status();
 }
 
+/******************** SYEVJ/HEEVJ ********************/
+hipsolverStatus_t hipsolverDnSsyevj_bufferSize(hipsolverDnHandle_t  handle,
+                                               hipsolverEigMode_t   jobz,
+                                               hipsolverFillMode_t  uplo,
+                                               int                  n,
+                                               float*               A,
+                                               int                  lda,
+                                               float*               D,
+                                               int*                 lwork,
+                                               hipsolverSyevjInfo_t params)
+try
+{
+    if(!handle)
+        return HIPSOLVER_STATUS_NOT_INITIALIZED;
+    if(lwork == nullptr)
+        return HIPSOLVER_STATUS_INVALID_VALUE;
+
+    *lwork = 0;
+    size_t sz;
+
+    rocblas_start_device_memory_size_query((rocblas_handle)handle);
+    hipsolverStatus_t status = rocblas2hip_status(rocsolver_ssyevd((rocblas_handle)handle,
+                                                                   hip2rocblas_evect(jobz),
+                                                                   hip2rocblas_fill(uplo),
+                                                                   n,
+                                                                   nullptr,
+                                                                   lda,
+                                                                   nullptr,
+                                                                   nullptr,
+                                                                   nullptr));
+    rocblas_stop_device_memory_size_query((rocblas_handle)handle, &sz);
+
+    size_t size_E = 0;
+
+    // space for E array
+    if(n > 0)
+        size_E = sizeof(float) * n;
```

Comment:
This is purely style, but I would use the conditional operator for this. The core difference between `if` and `?` is that the latter is an expression that returns a value, so my rule of thumb is to use `?` whenever I need a single value that depends on the result of a condition. I reserve `if` for when I don't need an expression because I'm calling functions without return values, or when there's multiple values that depend on a condition.

    size_t size_E = n > 0 ? sizeof(float) * n : 0;

That said, it's a matter of personal preference. If there's a reason you prefer this way, I'm totally fine with that.